src/PushCommand.php

```diff
@@ -410,10 +472,8 @@ private function getPackageName(InputInterface $input = null)
      */
     private function getNexusExtra($parameter, $default = null)
     {
-        $extras = $this->getComposer(true)->getPackage()->getExtra();
-
-        if (!empty($extras['nexus-push'][$parameter])) {
-            return $extras['nexus-push'][$parameter];
+        if (!empty($this->nexusPushConfig[$parameter])) {
```

Comment:
As explained above, you should not rely on the `nexusPushConfig` variable here, you should call the function `parseNexusExtra` every time and do the caching in the `parseNexusExtra` function directly.

Reply:
I don't quite understand. You mean call parseNexusExtra in getNexusExtra? Function parseNexusExtra requires the parameter InputInterface $input, which is expensive to add for getNexusExtra. Using $this->nexuspushconfig is much less risky

Reply:
Yes, I agree it's not an easy refactoring. See the comment above for more detail, a simple comment on top of this comment will do the trick for now.
I might do some refactoring for this whole part in the future.